galley/pkg/config/resource/name.go

```diff
@@ -29,8 +32,11 @@ func NewName(namespace, local string) Name {
 }
 
 // NewFullName returns a given name as a resource Name
-func NewFullName(name string) Name {
-	return Name{string: name}
+func NewFullName(name string) (Name, error) {
+	if name == "" || !strings.Contains(name, "/") {
```

Comment:
valid names may not contain a namespace. We had instances of this for cluster-scoped resources.

Reply:
The one main rule is, no name segment between `/`s can be empty. It is valid to have a name with a single segment.

Reply:
@ozevren updated the validation logic, see if it captures what you meant.